tests/c_api_tests/test_cpp_api_meta_data.cpp

```diff
@@ -0,0 +1,73 @@
+            CHECK(MetaData::get_dataset_by_name(dataset_name) == dataset);
+            CHECK(dataset_name == power_grid_model::meta_data::meta_data_gen::meta_data.datasets[idx_dataset].name);
+
+            // check component
+            power_grid_model::meta_data::MetaDataset const& cpp_dataset =
+                power_grid_model::meta_data::meta_data_gen::meta_data.get_dataset(dataset_name);
+            CHECK(MetaData::n_components(dataset) == cpp_dataset.n_components());
+            for (Idx idx_component = 0; idx_component != cpp_dataset.n_components(); ++idx_component) {
+                MetaComponent const* const component = MetaData::get_component_by_idx(dataset, idx_component);
+                std::string const component_name = MetaData::component_name(component);
+                CHECK(MetaData::get_component_by_name(dataset_name, component_name) == component);
+                power_grid_model::meta_data::MetaComponent const& cpp_component = cpp_dataset.components[idx_component];
+                CHECK(component_name == cpp_component.name);
+                CHECK(MetaData::component_size(component) == cpp_component.size);
+                CHECK(MetaData::component_alignment(component) == cpp_component.alignment);
+
+                // check attribute
+                CHECK(MetaData::n_attributes(component) == cpp_component.n_attributes());
+                for (Idx idx_attribute = 0; idx_attribute != cpp_component.n_attributes(); ++idx_attribute) {
+                    MetaAttribute const* const attribute = MetaData::get_attribute_by_idx(component, idx_attribute);
+                    std::string const attribute_name = MetaData::attribute_name(attribute);
+                    CHECK(MetaData::get_attribute_by_name(dataset_name, component_name, attribute_name) == attribute);
+                    power_grid_model::meta_data::MetaAttribute const& cpp_attribute =
+                        cpp_component.attributes[idx_attribute];
+                    CHECK(attribute_name == cpp_attribute.name);
+                    CHECK(MetaData::attribute_ctype(attribute) == static_cast<Idx>(cpp_attribute.ctype));
+                    CHECK(MetaData::attribute_offset(attribute) == cpp_attribute.offset);
+                }
+            }
+        }
+
+        SUBCASE("Endian") {
+            CHECK(static_cast<bool>(MetaData::is_little_endian()) == power_grid_model::meta_data::is_little_endian());
+        }
+
+        SUBCASE("Check error handling for unknown name") {
+            using namespace std::string_literals;
+
+            try {
+                MetaData::get_attribute_by_name("No_dataset", "no_name", "no attribute");
+            } catch (PowerGridRegularError const& e) {
+                std::string const err_msg{e.what()};
+                CHECK(err_msg.find("You supplied wrong name and/or index!"s) != std::string::npos);
+            }
+        }
+    }
+}
+
+} // namespace power_grid_model_cpp
```

Comment:
nitpick newline at EOF (please also check other files

```suggestion
} // namespace power_grid_model_cpp

```